### PR TITLE
Add missing docs for some PKCS12 functions

### PR DESCRIPTION
**Pull request by jon-oracle**
Some functions listed in include/openssl/pkcs12.h are missing documentation.

- [X] documentation is added or updated

### CODE REVIEW

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 1]
```diff
@@ -0,0 +1,44 @@
+=pod
+
+=head1 NAME
+
+PKCS12_SAFEBAG_get0_attrs,PKCS12_get_attr_gen - Retrieve attributes from a PKCS#12 safeBag
```
Comment (InfoHunter) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: It seems there should be a space after the comma.

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+=head1 RETURN VALUES
+
+PKCS12_SAFEBAG_get0_attrs() returns the stack of B<X509_ATTRIBUTE>s from a PKCS#12 safeBag, which could be empty.
+
+PKCS12_get_attr_gen() returns an ASN1_TYPE object containing the attribute, or NULL if the attribute was not present or an error occurred.
+PKCS12_get_attr_gen() does not allocate a new attribute. The returned attribute is still owned by the B<PKCS12_SAFEBAG> in which it resides.
+
+=head1 SEE ALSO
+
+L<PKCS12_add_friendlyname_asc(3)>,
```
Comment (InfoHunter) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Should be no comma at the end of line?

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+ const STACK_OF(X509_ATTRIBUTE) *PKCS12_SAFEBAG_get0_attrs(const PKCS12_SAFEBAG *bag);
+
+ ASN1_TYPE *PKCS12_get_attr_gen(const STACK_OF(X509_ATTRIBUTE) *attrs,
+                                int attr_nid)
+
+=head1 DESCRIPTION
+
+PKCS12_SAFEBAG_get0_attrs() retrieves the stack of B<X509_ATTRIBUTE>s from a PKCS#12 safeBag.
+B<bag> is the B<PKCS12_SAFEBAG> retrieve the attributes from.
```
Comment (InfoHunter) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: This sentence sounds not very natural...

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+PKCS12_get_attr_gen() returns an ASN1_TYPE object containing the attribute, or NULL if the attribute was not present or an error occurred.
+PKCS12_get_attr_gen() does not allocate a new attribute. The returned attribute is still owned by the B<PKCS12_SAFEBAG> in which it resides.
+
+=head1 SEE ALSO
+
+L<PKCS12_add_friendlyname_asc(3)>,
+
+=head1 COPYRIGHT
+
+Copyright 2002-2019 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (InfoHunter) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: This should be 2019 only, but it seems `make update` could fix this as well.

Comment (levitte) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: I don't see how any update would make it 2019 only...

Comment (InfoHunter) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Ahh, I recall there must be some scripts that can update this date 'up to date' when releasing a new version ;-)

**Comment by mspncp** [comment]
https://github.com/openssl/openssl/blob/master/util/openssl-update-copyright


```
util/openssl-update-copyright —help
```

**Comment by mspncp** [comment]
The script updates only the end year. For new files, just specify 2019, no range.

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod (hunk 1 above)
Comment (jon-oracle) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Fixed this and another occurrence.

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod (hunk 2 above)
Comment (jon-oracle) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Removed

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod (hunk 3 above)
Comment (jon-oracle) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Fixed

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod (hunk 4 above)
Comment (jon-oracle) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: This was some copypasta I left in. Fixed now.

Comment (mspncp) [reply doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: "copypasta" - Nice pun. Reminds me of "spaghetti code". I'll remember that one :laughing:

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+ const STACK_OF(X509_ATTRIBUTE) *PKCS12_SAFEBAG_get0_attrs(const PKCS12_SAFEBAG *bag);
+
+ ASN1_TYPE *PKCS12_get_attr_gen(const STACK_OF(X509_ATTRIBUTE) *attrs,
+                                int attr_nid)
+
+=head1 DESCRIPTION
+
+PKCS12_SAFEBAG_get0_attrs() retrieves the stack of B<X509_ATTRIBUTE>s from a PKCS#12 safeBag.
+B<bag> is the B<PKCS12_SAFEBAG> retrieve the attributes from.
```
Comment (slontis) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: 'to' retrieve

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+PKCS12_get_attr_gen() retrieves an attribute by NID from a stack of B<X509_ATTRIBUTE>s.
+B<attr_nid> is the NID for the attribute to retrieve.
```
Comment (slontis) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: replace for with of?

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+=head1 RETURN VALUES
+
+PKCS12_SAFEBAG_get0_attrs() returns the stack of B<X509_ATTRIBUTE>s from a PKCS#12 safeBag, which could be empty.
+
+PKCS12_get_attr_gen() returns an ASN1_TYPE object containing the attribute, or NULL if the attribute was not present or an error occurred.
```
Comment (slontis) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: was 'either' not present

File: doc/man3/PKCS12_get_friendlyname.pod [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+=head1 DESCRIPTION
+
+PKCS12_get_friendlyname() retrieves a UTF-8 string representation of the PKCS#9 friendlyName attribute for a PKCS#12 safeBag item.
+
+B<bag> is the B<PKCS12_SAFEBAG> to retrieve the attribute from.
+
+=head1 RETURN VALUES
+
+A UTF-8 string, or NULL if the attribute was not present or an error occurred.
```
Comment (slontis) [inline doc/man3/PKCS12_get_friendlyname.pod]: if either

**Comment by slontis** [comment]
Congrats - you have been NITTED :)

**Comment by jon-oracle** [comment]
Anything else needed for this?

**Comment by slontis** [comment]
ping

File: doc/man3/PKCS12_add_localkeyid.pod [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+ #include <openssl/pkcs12.h>
+
+ int PKCS12_add_localkeyid(PKCS12_SAFEBAG *bag, const char *name,
+                                int namelen);
+
+=head1 DESCRIPTION
+
+PKCS12_add_localkeyid() adds an octet string representation of the PKCS#9 localKeyId attribute to a PKCS#12 safeBag.
+
+B<bag> is the B<PKCS12_SAFEBAG> to add the attribute to.
```
Comment (t8m) [inline doc/man3/PKCS12_add_localkeyid.pod]: as above

File: doc/man3/PKCS12_get_friendlyname.pod [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+ #include <openssl/pkcs12.h>
+
+ char *PKCS12_get_friendlyname(PKCS12_SAFEBAG *bag);
+
+=head1 DESCRIPTION
+
+PKCS12_get_friendlyname() retrieves a UTF-8 string representation of the PKCS#9 friendlyName attribute for a PKCS#12 safeBag item.
+
+B<bag> is the B<PKCS12_SAFEBAG> to retrieve the attribute from.
```
Comment (t8m) [inline doc/man3/PKCS12_get_friendlyname.pod]: as above

Comment (jon-oracle) [reply doc/man3/PKCS12_get_friendlyname.pod]: Fixed these nits

File: doc/man3/PKCS12_add_friendlyname_asc.pod [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+ #include <openssl/pkcs12.h>
+
+ int PKCS12_add_friendlyname_asc(PKCS12_SAFEBAG *bag, const char *name,
+                                int namelen);
+
+ int PKCS12_add_friendlyname_utf8(PKCS12_SAFEBAG *bag, const char *name,
+                                int namelen);
+
+ int PKCS12_add_friendlyname_uni(PKCS12_SAFEBAG *bag,
+                                const unsigned char *name, int namelen);
```
Comment (mattcaswell) [inline doc/man3/PKCS12_add_friendlyname_asc.pod]: Nit: looks like there is an indentation issue here. The params on each of the second lines above should line up with the start of the params list on the line before.

File: doc/man3/PKCS12_SAFEBAG_get0_attrs.pod [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+PKCS12_get_attr_gen() retrieves an attribute by NID from a stack of B<X509_ATTRIBUTE>s.
+I<attr_nid> is the NID of the attribute to retrieve.
+
+=head1 RETURN VALUES
+
+PKCS12_SAFEBAG_get0_attrs() returns the stack of B<X509_ATTRIBUTE>s from a PKCS#12 safeBag, which could be empty.
+
+PKCS12_get_attr_gen() returns an B<ASN1_TYPE> object containing the attribute, or NULL if the attribute was either not present or an error occurred.
+PKCS12_get_attr_gen() does not allocate a new attribute. The returned attribute is still owned by the B<PKCS12_SAFEBAG> in which it resides.
```
Comment (mattcaswell) [inline doc/man3/PKCS12_SAFEBAG_get0_attrs.pod]: Please line break at 80.

File: doc/man3/PKCS12_add_friendlyname_asc.pod [hunk 13]
```diff
@@ -0,0 +1,45 @@
+=pod
+
+=head1 NAME
+
+PKCS12_add_friendlyname_asc, PKCS12_add_friendlyname_utf8, PKCS12_add_friendlyname_uni - Functions to add the friendlyname attribute to a PKCS#12 safeBag
```
Comment (mattcaswell) [inline doc/man3/PKCS12_add_friendlyname_asc.pod]: Line break

File: doc/man3/PKCS12_add_friendlyname_asc.pod [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+                                int namelen);
+
+ int PKCS12_add_friendlyname_uni(PKCS12_SAFEBAG *bag,
+                                const unsigned char *name, int namelen);
+
+=head1 DESCRIPTION
+
+PKCS12_add_friendlyname_asc() adds an ASCII string representation of the PKCS#9 friendlyName attribute to a PKCS#12 safeBag.
+PKCS12_add_friendlyname_utf8() adds a UTF-8 string representation of the PKCS#9 friendlyName attribute to a PKCS#12 safeBag.
+PKCS12_add_friendlyname_uni() adds a Unicode string representation of the PKCS#9 friendlyName attribute to a PKCS#12 safeBag.
```
Comment (mattcaswell) [inline doc/man3/PKCS12_add_friendlyname_asc.pod]: Line break at 80

File: doc/man3/PKCS12_add_localkeyid.pod [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+=head1 NAME
+
+PKCS12_add_localkeyid - Add the localKeyId attribute to a PKCS#12 safeBag
+
+=head1 SYNOPSIS
+
+ #include <openssl/pkcs12.h>
+
+ int PKCS12_add_localkeyid(PKCS12_SAFEBAG *bag, const char *name,
+                                int namelen);
```
Comment (mattcaswell) [inline doc/man3/PKCS12_add_localkeyid.pod]: Indentation

File: doc/man3/PKCS12_get_friendlyname.pod [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+=head1 SYNOPSIS
+
+ #include <openssl/pkcs12.h>
+
+ char *PKCS12_get_friendlyname(PKCS12_SAFEBAG *bag);
+
+=head1 DESCRIPTION
+
+PKCS12_get_friendlyname() retrieves a UTF-8 string representation of the PKCS#9 friendlyName attribute for a PKCS#12 safeBag item.
```
Comment (mattcaswell) [inline doc/man3/PKCS12_get_friendlyname.pod]: line break

**Comment by jon-oracle** [comment]
ping

**Comment by t8m** [comment]
ping @openssl/omc

File: doc/man3/PKCS12_get_friendlyname.pod [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+PKCS12_get_friendlyname() retrieves a UTF-8 string representation of the PKCS#9
+friendlyName attribute for a PKCS#12 safeBag item.
+
+I<bag> is the B<PKCS12_SAFEBAG> to retrieve the attribute from.
+
+=head1 RETURN VALUES
+
+A UTF-8 string, or NULL if the attribute was either not present or an error occurred.
+
+The returned string is allocated by OpenSSL and should be freed by the user.
```
Comment (mattcaswell) [inline doc/man3/PKCS12_get_friendlyname.pod]: This doesn't seem to be true.

Comment (mattcaswell) [reply doc/man3/PKCS12_get_friendlyname.pod]: Ignore me...I read what you wrote the wrong way around,.

**Comment by t8m** [comment]
Squashed the commits and merged to master as eb389a75

**Pull request closed without merging**